File: Renderer/src/main/java/org/turbomedia/turboedit/renderer/Queue.java
package org.turbomedia.turboedit.renderer;

import java.util.HashMap;
import java.util.UUID;

public class Queue {
    private static HashMap<String, QueueItem> queue = new HashMap<>();

    public static void AddToQueue(QueueItem item) {
        var id = UUID.randomUUID().toString();
        queue.put(id, item);
    }

    public static QueueItem GetFromQueue(String id) {
        return queue.get(id);
    }

    public static void RemoveFromQueue(String id) {
        queue.remove(id);
    }

    public record QueueItem(){}
}

File: Renderer/src/main/kotlin/turboedit/renderer/Queue.kt
package turboedit.renderer

import java.util.HashMap
import java.util.UUID

object Queue {
    private val queue = HashMap<String?, QueueItem?>()

    fun addToQueue(item: QueueItem?) {
        val id: String? = UUID.randomUUID().toString()
        queue.put(id, item)
    }

    fun getFromQueue(id: String?): QueueItem? {
        return queue.get(id)
    }

    fun removeFromQueue(id: String?) {
        queue.remove(id)
    }

    class QueueItem
}